src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,495 @@
+                The name to give the data asset.
+            * tags : Optional[List[str]]
+                The tags to use to describe the data asset.
+            * custom_metadata : Optional[dict]
+                What key:value metadata tags to apply to the asset.
+            * viewable_to_everyone : bool
+                Whether to share the captured results with everyone.
+
+        Returns
+        -------
+        requests.Response
+
+        """
+        if capture_result_config.asset_name is None:
+            assert (
+                input_data_asset_name is not None
+            ), "Either asset_name or input_data_asset_name must be provided"
+            capture_time = datetime_to_name_string(datetime.now())
+            capture_result_config.asset_name = (
+                f"{input_data_asset_name}"
+                f"_{capture_result_config.process_name}"
+                f"_{capture_time}"
+            )
+
+        if capture_result_config.mount is None:
+            capture_result_config.mount = capture_result_config.asset_name
+
+        computation_source = Sources.Computation(
+            id=computation_id,
+        )
+        source = Source(computation=computation_source)
+        create_data_asset_request = CreateDataAssetRequest(
+            name=capture_result_config.asset_name,
+            tags=capture_result_config.tags,
+            mount=capture_result_config.mount,
+            source=source,
+            custom_metadata=capture_result_config.custom_metadata,
+        )
+
+        reg_result_response = self.co_client.create_data_asset(
+            create_data_asset_request
+        )
+        registered_results_response_json = reg_result_response.json()
+
+        # TODO: This step intermittently breaks. Adding extra check to help
+        #  figure out why.
+        if registered_results_response_json.get("id") is None:
+            raise KeyError(
+                f"Something went wrong registering"
+                f" {capture_result_config.asset_name}. "
+                f"Response Status Code: {reg_result_response.status_code}. "
+                f"Response Message: {registered_results_response_json}"
+            )
+
+        results_data_asset_id = registered_results_response_json["id"]
+        response_res_available = self._wait_for_data_availability(
+            data_asset_id=results_data_asset_id
+        )
+
+        if response_res_available.status_code != 200:
+            raise FileNotFoundError(f"Unable to find: {results_data_asset_id}")
+
+        # Make captured results viewable to everyone
+        if capture_result_config.viewable_to_everyone:
+            update_res_perm_response = self.co_client.update_permissions(
+                data_asset_id=results_data_asset_id, everyone="viewer"
+            )
+            logger.info(
+                f"Updating permissions {update_res_perm_response.status_code}"
+            )
+        return reg_result_response
+
+    def process(self) -> dict:
```

Comment:
Most of the other Job classes we've been writing have a public `run_job` method. It might be better for us to be consistent.